consensus/hotstuff/eventloop/event_loop_v2.go

```diff
@@ -61,7 +76,7 @@ func (el *EventLoopV2) loop() {
 	// if hotstuff hits any unknown error, it will exit the loop
 
 	for {
-		quitted := el.unit.Quit()
+		quitted := ctx.Done()
```

Comment:
"quitted"  sounds a bit funny to me. Maybe we could call it 
```suggestion
		shutdownSignaled := ctx.Done()
```
Furthermore, I believe we should move this out of the loop for performance reasons.